common_drivers/platform_drivers/platform_drivers.c

```diff
@@ -315,6 +316,7 @@ int32_t gpio_get(gpio_desc **desc,
 		return FAILURE;
 
 	if (gpio_number < 32) {
+		free(dev);
 		return FAILURE;
 	}
```

Comment:
in this case, if you move this code block before the malloc() you don't need to do the free()